R/lazy-join-query.R

```diff
@@ -196,15 +196,26 @@ sql_build.lazy_multi_join_query <- function(op, con, ...) {
 }
 
 generate_join_table_names <- function(table_names) {
+  table_name_length_max <- dplyr::coalesce(max(nchar(table_names$name)), 0)
+
   if (length(table_names$name) != 2) {
     table_names_repaired <- vctrs::vec_as_names(table_names$name, repair = "unique", quiet = TRUE)
     auto_name <- table_names$from != "as"
     table_names$name[auto_name] <- table_names_repaired[auto_name]
-
-    return(table_names$name)
+    table_names_prepared <- table_names$name
+  } else{
+    table_names_prepared <- join_two_table_alias(table_names$name, table_names$from)
   }
 
-  join_two_table_alias(table_names$name, table_names$from)
+  # avoid database aliases exceeding the database-specific maximum length
+  table_names_prepared %>%
+    abbreviate(
+      # arbitrarily floor at 63 (Postgres limit) to avoid unnecessarily truncating reasonable lengths
```

Comment:
Not ideal, but I couldn't think of a better way to _not_ do use an arbitrary floor.

For example, without, then: `name_LHS` and `name_RHS` were getting truncated to something like `nLHS` and `nRHS` which isn't informative / necessary.